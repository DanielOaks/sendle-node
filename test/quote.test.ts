import { describe, expect, it } from 'vitest';
import { client } from './helpers';

describe('Quote', () => {
  it('should return a quote', async () => {
    const quote = await client.quote({
      delivery_suburb: 'Erskineville',
      delivery_postcode: '2043',
      pickup_suburb: 'Surry Hills',
      pickup_postcode: '2010',
      weight_value: '1',
      weight_units: 'kg',
    });

    expect(quote).toMatchSnapshot({
      eta: {
        date_range: expect.arrayContaining([expect.any(String)]),
        days_range: expect.arrayContaining([expect.any(Number)]),
        for_pickup_date: expect.any(String),
      },
      plan_name: expect.any(String),
      price_breakdown: {
        base: {
          amount: expect.any(Number),
        },
        base_tax: {
          amount: expect.any(Number),
        },
        fuel_surcharge: {
          amount: expect.any(Number),
        },
        fuel_surcharge_tax: {
          amount: expect.any(Number),
        },
      },
      tax_breakdown: {
        gst: {
          amount: expect.any(Number),
        },
      },
      quote: {
        gross: {
          amount: expect.any(Number),
        },
        net: {
          amount: expect.any(Number),
        },
        tax: {
          amount: expect.any(Number),
        },
      },
    });
  });
});
