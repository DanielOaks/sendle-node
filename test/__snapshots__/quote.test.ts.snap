// Vitest Snapshot v1

exports[`Quote > should return a quote 1`] = `
{
  "allowed_packaging": "any",
  "eta": {
    "date_range": ArrayContaining [
      Any<String>,
    ],
    "days_range": ArrayContaining [
      Any<Number>,
    ],
    "for_pickup_date": Any<String>,
  },
  "plan_name": Any<String>,
  "price_breakdown": {
    "base": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "base_tax": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "cover": {
      "amount": 0,
      "currency": "AUD",
    },
    "cover_tax": {
      "amount": 0,
      "currency": "AUD",
    },
    "discount": {
      "amount": 0,
      "currency": "AUD",
    },
    "discount_tax": {
      "amount": 0,
      "currency": "AUD",
    },
    "fuel_surcharge": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "fuel_surcharge_tax": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
  },
  "quote": {
    "gross": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "net": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
    "tax": {
      "amount": Any<Number>,
      "currency": "AUD",
    },
  },
  "route": {
    "description": "Sydney, Australia to Sydney, Australia",
    "type": "same-city",
  },
  "tax_breakdown": {
    "gst": {
      "amount": Any<Number>,
      "currency": "AUD",
      "rate": 0.1,
    },
  },
}
`;
